rtic-monotonics/src/nrf/rtc.rs

```diff
@@ -0,0 +1,241 @@
+                // basepri/source masking based critical sections.
+                unsafe {
+                    crate::set_monotonic_prio(pac::NVIC_PRIO_BITS, Interrupt::$rtc);
+                    pac::NVIC::unmask(Interrupt::$rtc);
+                }
+            }
+
+            /// Used to access the underlying timer queue
+            #[doc(hidden)]
+            pub fn __tq() -> &'static TimerQueue<$mono_name> {
+                &$tq
+            }
+
+            /// Timeout at a specific time.
+            #[inline]
+            pub async fn timeout_at<F: Future>(
+                instant: <Self as Monotonic>::Instant,
+                future: F,
+            ) -> Result<F::Output, TimeoutError> {
+                $tq.timeout_at(instant, future).await
+            }
+
+            /// Timeout after a specific duration.
+            #[inline]
+            pub async fn timeout_after<F: Future>(
+                duration: <Self as Monotonic>::Duration,
+                future: F,
+            ) -> Result<F::Output, TimeoutError> {
+                $tq.timeout_after(duration, future).await
+            }
+
+            /// Delay for some duration of time.
+            #[inline]
+            pub async fn delay(duration: <Self as Monotonic>::Duration) {
+                $tq.delay(duration).await;
+            }
+
+            /// Delay to some specific time instant.
+            #[inline]
+            pub async fn delay_until(instant: <Self as Monotonic>::Instant) {
+                $tq.delay_until(instant).await;
+            }
+
+            #[inline(always)]
+            fn is_overflow() -> bool {
+                let rtc = unsafe { &*$rtc::PTR };
+                rtc.events_ovrflw.read().bits() == 1
+            }
+        }
+
+        #[cfg(feature = "embedded-hal-async")]
+        impl embedded_hal_async::delay::DelayUs for $mono_name {
+            type Error = core::convert::Infallible;
+
+            #[inline]
+            async fn delay_us(&mut self, us: u32) -> Result<(), Self::Error> {
+                $tq.delay((us as u64).micros()).await;
+                Ok(())
+            }
+
+            #[inline]
+            async fn delay_ms(&mut self, ms: u32) -> Result<(), Self::Error> {
+                $tq.delay((ms as u64).millis()).await;
+                Ok(())
+            }
+        }
+
+        impl Monotonic for $mono_name {
+            const ZERO: Self::Instant = Self::Instant::from_ticks(0);
+
+            type Instant = fugit::TimerInstantU64<32_768>;
+            type Duration = fugit::TimerDurationU64<32_768>;
+
+            fn now() -> Self::Instant {
+                // In a critical section to not get a race between overflow updates and reading it
+                // and the flag here.
+                critical_section::with(|_| {
+                    let rtc = unsafe { &*$rtc::PTR };
+                    let cnt = rtc.counter.read().bits() as u64;
+                    let ovf = if Self::is_overflow() {
```

Comment:
if overflow happens right between these lines, it'll give the wrong result

```rust
let cnt = rtc.counter.read().bits() as u64;
// OVERFLOW HAPPENS HERE
let ovf = if Self::is_overflow() {
```

Reply:
So, a fix might be to re-read the `cnt` value on overflow.